runelite-client/src/main/java/net/runelite/client/plugins/timetracking/farming/PaymentTracker.java

```diff
@@ -90,6 +100,48 @@ private static String configKey(FarmingPatch fp)
 		return fp.configKey() + "." + TimeTrackingConfig.PROTECTED;
 	}
 
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked menuOptionClicked)
+	{
+		switch (menuOptionClicked.getMenuAction())
+		{
+			case NPC_THIRD_OPTION:
+				chosenPatch = 1;
+				break;
+			case NPC_FOURTH_OPTION:
+				chosenPatch = 2;
+				break;
+			case WIDGET_CONTINUE:
+				chosenPatch = (menuOptionClicked.getActionParam() != 1 && menuOptionClicked.getActionParam() != 2) ? -1 : menuOptionClicked.getActionParam();
+				break;
+			default:
+				chosenPatch = -1;
+		}
+	}
+
+	@Subscribe
+	public void onScriptPreFired(ScriptPreFired event)
+	{
+		if (event.getScriptId() != ScriptID.KEYINPUT_MATCHED)
+		{
+			return;
+		}
+
+		int pressedKey = client.getIntStack()[1];
+		int componentID = client.getIntStack()[0];
+
+		if ((pressedKey == 1 || pressedKey == 2) && widgetActive)
```

Comment:
I don't see what `widgetActive` is doing here. Why can't I delete it and have the code work better because it then doesn't require me to keypress Continue on some previous npc dialog?